fishtank/src/node.ts

```diff
@@ -28,19 +30,28 @@ export class Node {
     return join(tmpdir(), 'fishtank', this.cluster.name, this.name, '.ironfish')
   }
 
-  get ipcSocketPath(): string {
-    return join(this.dataDir, 'ironfish.ipc')
+  async getRpcTcpPort(): Promise<number> {
+    const info = await this.backend.inspect(this.containerName)
+    const port = info.ports?.tcp.get(INTERNAL_RPC_TCP_PORT)
+    if (port) {
+      return port
+    } else {
+      throw new Error('Node is not exposing any RPC port')
+    }
   }
 
-  async connectRpc(): Promise<RpcClient> {
+  async connectRpc(): Promise<RpcSocketClient> {
+    const rpcTcpPort = await this.getRpcTcpPort()
     const sdk = await IronfishSdk.init({
       dataDir: this.dataDir,
       configOverrides: {
-        enableRpcTcp: false,
-        ipcPath: this.ipcSocketPath,
+        enableRpcTcp: true,
+        enableRpcIpc: false,
+        enableRpcTls: false,
```

Comment:
Why do you have tls set to false?

Reply:
This simulator is meant to be used in local (trusted) setups, so I do not see much value in using TLS. Using TLS carries the problem that the authentication token is required to use the RPC. In the original version of this code, TLS was enabled, and the authentication token was hardcoded to a fixed value.

Hardcoding the authentication token (or even generating it on the fly, and storing it somewhere) adds some complexity to the code. It's not a lot of complexity, but still it's a bunch of extra code, and when I looked at it, I did not see a valid reason to keep that extra code. Disabling TLS (and therefore disabling the auth token) seems to me the best thing to do in terms of simplicity.

Again, the assumption here is that security is not a concern because we're running in a trusted environment, but if there are different opinions about this I'm all ears.

Reply:
The default behavior when connecting via TCP is with TLS. We are bypassing that part of the node code, which seems like a reasonable tradeoff, but just misses a small fraction of code on the canonical path:
https://github.com/iron-fish/ironfish/blob/1bf0c8ed7e8eb63839af160cb1d076a564e104af/ironfish/src/rpc/clients/tlsClient.ts

Seems unlikely there will be major performance changes based on w vs w/o TLS.